### DISCUSSION TITLE
Confused about dimension order in MONAI and how to develop C++ inference with ITK

### DISCUSSION THREAD

**Discussion by 7oud** [Project-MONAI/MONAI#2888] [General]
When using the `LoadImage` transform, both `NibabelReader` and `ITKReader` load nifti file as `WHD` data order, even ITKReader does a transpose for it, Although this dimension order is the same with some meta info in dicom, like spacing. But not for the best performance. Now, I use the MONAI trained model with ITK libraries in C++, it's very inconvenient. If using ITK functions, the data dimension order must be `DHW`, but when using model to inference, I have to tranpose the image buffer(data) to `WHD`, after inference, then tranpose the outputs back to `DHW` and write to the ITK image buffer. Specifically if using slide window inference,
the additional two tranpose and `WHD` order inference affect performance a lot. Is there some good methods to deal with this situation. Or can I train models with `DHW` data order with MONAI?

**Comment by 7oud** [comment]
And MONAI is PyTorch-based, PyTorch use the `DHW` order, why MONAI doesn't choose the same one?

**Comment by wyli** [comment]
which version of MONAI is this related to? asking because recently there are some API enhancements in this area in both monai and itk-python

**Comment by 7oud** [comment]
The lastest `monai-weekly 0.7.dev2135 ` and `monai 0.6`, I dont find the method to train a `DHW` dim order using these 2 versions
